### DISCUSSION TITLE
Upload custom icon

### DISCUSSION THREAD

**Discussion by DL-TS2024** [directus/directus#23157] [Feature Requests]
### Summary

Unable to upload and specify custom SVG icons.

### Basic Example

The material icon set is quite comprehensive, but there maybe company or brand specific icons which already have significant meaning for users which are not in the included icon set. It would therefore be great to be able to upload and use custom icons alongside the material ones.

### Motivation

- Easier user adoption for those who are not familiar (developers and users) with the material icon set.
- New options would be available in the icon field picker too, so may open up extended use cases using icons when storing data, or a pictorial representation.

### Detailed Design

- Location in the file library explicitly for icons
- Guidelines on size of icon for the platform
- Ability to select the icons uploaded into the file library location in the icon picker

### Must Have:
- Ability to select from file library location
- Included alongside the existing icons

### Should Have:
- Guideline on expected sizing

### Could Have:
- Option for multiple names to help users locate icons

### Won't Have:
- Colour picker

### Drawbacks

Higher priority development items.

### Alternatives

No alternatives.

### Adoption Strategy

Unknown

**Comment by shadiramadan** [comment]
It would be REALLY nice to support other icon sets + custom icons.
I've partially worked around things by uploading my icons to files and using the file title to select the icon in my app UI.  I only rely on the title when querying the API but directus can use the file itself for displaying the Icon in the directus UI

**Comment by jonaskohl** [comment]
I would suggest migrating the whole icon system over to SVGs identified by a name (instead of relying on icons fonts, [which shouldn't be used anymore](https://www.irigoyen.dev/blog/2021/02/17/stop-using-icon-fonts/)). This would ensure backward compatibility by offering all Material *Icons* (not *Symbols*) under their current names while also allowing users (or at least extensions) to extend this system icon library by registering their own icons (maybe even allow icon namespacing to avoid naming conflicts). I think icons should *not* use the Directus files system and instead use inline SVGs directly (to ensure the best visual quality while also allowing them to be easily recolored and styled via CSS &ndash; important for theming!)

**Comment by salva-dev** [comment]
Yes, please. Good feature. My company use fontawesome.
